Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,408 @@
+                        log.warning('Warning: mean value in selected roi is more than 10 percent of data range - may not represent the background')
+
+                self.roi_slice = slc
+                self.roi_axes = axes
+                
+            else:
+                raise TypeError("roi must be a dictionary, found {}"
+                .format(str(type(self.roi))))
+        
+        
+
+        # convert flux array to float32
+        self.flux = numpy.array(self.flux, dtype=numpy.float32, ndmin=1)
+
+        # check flux array is the right size
+        flux_size_flat = len(self.flux.ravel())
+        if flux_size_flat > 1:
+            data_size_flat = len(dataset.geometry.angles)*dataset.geometry.channels
+            if data_size_flat != flux_size_flat:
+                raise ValueError("Flux must be a scalar or array with length \
+                                    \n = number of projections, found {} and {}"
+                                    .format(flux_size_flat, data_size_flat))
+            
+        # check if flux array contains 0s
+        if 0 in self.flux:
+            raise ValueError('Flux value can\'t be 0, provide a different flux\
+                                or region of interest with non-zero values')
+          
+    def _calculate_target(self):
+        '''
+        Calculate the target value for the normalisation
+        '''
+
+        if self.flux is None:
+            raise ValueError('Flux not found')
+            
+        if isinstance(self.target, (int,float)):
+            self.target_value = self.target
+        elif isinstance(self.target, str):
+            if self.target == 'first':
+                if len(numpy.shape(self.flux)) > 0 :
+                    self.target_value = self.flux.flat[0]
+                else:
+                    self.target_value = self.flux
+            elif self.target == 'mean':
+                self.target_value = numpy.mean(self.flux.ravel())
+            else:
+                raise ValueError("Target string not recognised, found {}, expected 'first' or 'mean'"
+                                 .format(self.target))
+        else:
+            raise TypeError("Target must be string or a number, found {}"
+                            .format(type(self.target)))
+            
+    def preview_configuration(self, angle=None, channel=None, log=False):
+        '''
+        Preview the FluxNormalisation processor configuration for roi mode.
+        Plots the region of interest on the image and the mean, maximum and 
+        minimum intensity in the roi.
+        
+        Parameters:
+        -----------
+        angle: float, optional
+            Angle to plot, default=None displays the data with the minimum
+            and maximum pixel values in the roi, otherwise the angle to display 
+            can be specified as a float and the closest angle will be displayed.
+            For 2D data, the roi is plotted on the sinogram.
```

Comment:
Elsewhere I think we use 'angle' as the name of the axis, then the argument is the index of that projection in a list, not the value. It might be worth just discussing this further, not because it's wrong but it could just be a change of expected behaviour. Would it cause any issues if they specified an index?

It would also have to be careful about units - especially if we decide to convert things in the future and store in a defined way.

Reply:
Thanks Gemma, I've updated this now so it uses the index and I've also updated the preview_configuration plot so it shows the angle and the index